.github/workflows/ci.yaml

```diff
@@ -143,12 +143,25 @@ jobs:
           enableCrossOsArchive: true
           save-always: true
 
+      # Run all tests in *.py files, which excludes tests in *.yml files that test type
+      # annotations via the pytest-mypy-plugins plugin, because the type annotations
+      # tests fail when using multiple processes (the -n option to pytest).
       - name: Run tests
         run: python -m pytest -n 4
           --timeout 180
           --cov=xarray
           --cov-report=xml
           --junitxml=pytest.xml
+          xarray/tests/test_*.py
+
+      # As noted in the comment on the previous step, we must run type annotation tests
+      # separately, and we will run them even if the preceding tests failed.  Further,
+      # we must restrict these tests to run only when matrix.env is empty, as this is
+      # the only case when all of the necessary dependencies are included such that
+      # spurious mypy errors due to missing packages are eliminated.
+      - name: Run mypy tests
+        if: ${{ always() && matrix.env == '' }}
+        run: python -m pytest xarray/tests/test_*.yml
```

Comment:
I'm quite committed to ensuring that running a simple `pytest` works! I really think it's important that running tests doesn't require looking up various commands in `.github/workflows/ci.yaml`

If we want to have tests that don't run by default, then we can use pytest marks and run them with an additional options; check out `--run-nightly` for an example...

Reply:
It took a bit of digging, but I _think_ I've made the changes you're looking for. If not, please provide specific guidance.

Reply:
Looks great, thank you v much